test/random.jl

```diff
@@ -537,3 +537,13 @@ let g = Base.Random.GLOBAL_RNG,
     @test srand(m, rand(UInt32, rand(1:10))) === m
     @test srand(m, rand(1:10)) === m
 end
+
+# Issue 20062 - ensure internal functions reserve_1, reserve are type-stable
+let r = MersenneTwister(0)
+    @test  Base.Random.mt_empty(r); @inferred Base.Random.reserve_1(r)
+    @test !Base.Random.mt_empty(r); @inferred Base.Random.reserve_1(r)
+
+    rand(r)
+    @test 0 < Base.Random.mt_avail(r) < Base.Random.MTCacheLength; @inferred Base.Random.reserve(r, Base.Random.mt_avail(r))
```

Comment:
I was probably being overly-cautious with these tests, trying to test inferred return type on different conditions of `mt_avail`. Probably sufficient to test once only. Will update to simplify.